numba_cuda/numba/cuda/dispatcher.py

```diff
@@ -830,12 +831,20 @@ def call(self, args, griddim, blockdim, stream, sharedmem):
         """
         Compile if necessary and invoke this kernel with *args*.
         """
-        if self.specialized:
-            kernel = next(iter(self.overloads.values()))
-        else:
-            kernel = _dispatcher.Dispatcher._cuda_call(self, *args)
+        with launch_config_ctx(
+            dispatcher=self,
+            args=args,
+            griddim=griddim,
+            blockdim=blockdim,
+            stream=stream,
+            sharedmem=sharedmem,
+        ):
+            if self.specialized:
```

Comment:
Specialized kernels cannot be recompiled, so a new launch configuration would not be able to affect the compilation of a new version - so this check could be kept outside the context manager.